src/mlpack/methods/reinforcement_learning/environment/acrobat.hpp

```diff
@@ -0,0 +1,317 @@
+   * @param linkLength1 length of link 1.
+   * @param linkLength2 length of link 2.
+   * @param linkMass1 mass of link 1.
+   * @param linkMass2 mass of link 2.
+   * @param linkCom1 position of the center of mass of link 1.
+   * @param linkCom2 position of the center of mass of link 2.
+   * @param linkMoi moments of inertia for both link.
+   * @param maxVel1 max angular velocity of link1.
+   * @param maxVel2 max angular velocity of link2.
+   */
+  Acrobat(const double gravity = 9.81,
+          const double linkLength1 = 1.0,
+          const double linkLength2 = 1.0,
+          const double linkMass1 = 1.0,
+          const double linkMass2 = 1.0,
+          const double linkCom1 = 0.5,
+          const double linkCom2 = 0.5,
+          const double linkMoi = 1.0,
+          const double maxVel1 = 4*M_PI,
+          const double maxVel2 = 9*M_PI,
+          const double dt = 0.2) :
+      gravity(gravity),
+      linkLength1(linkLength1),
+      linkLength2(linkLength2),
+      linkMass1(linkMass1),
+      linkMass2(linkMass2),
+      linkCom1(linkCom1),
+      linkCom2(linkCom2),
+      linkMoi(linkMoi),
+      maxVel1(maxVel1),
+      maxVel2(maxVel2),
+      dt(dt)
+  { /* Nothing to do here */ }
+  /**
+    * Dynamics of the Acrobat System.
+    * To get reward and next state based on current
+    * state and current action .
+    * 
+    * @param state The current State
+    * @param action_ The action taken
+    * @param nextState The next state
+    * return reward -1
+    * Always return -1 reward
+    */
+  double Sample(const State& state,
+                 const Action& action,
+                 State& nextState) const
+  {
+    arma::colvec state_ = {state.Theta1(), state.Theta2(),
+                          state.AngularVelocity1(),
+                          state.AngularVelocity2()};
+    double torque = Torque(action);
+    arma::colvec nextstate = Rk4(state_, torque);
+
+    nextState.Theta1() = Wrap(nextstate[0], -M_PI, M_PI);
+
+    nextState.Theta2() = Wrap(nextstate[1], -M_PI, M_PI);
+    nextState.AngularVelocity1() = Bound(nextstate[2],
+                                             -maxVel1 , maxVel1);
+    nextState.AngularVelocity2() = Bound(nextstate[3],
+                                             -maxVel2 , maxVel2);
+    return -1;
+  };
+  double Sample(const State& state, const Action& action) const
+  {
+    State nextState;
+    return Sample(state, action, nextState);
+  }
+  State InitialSample() const
+  {
+    return State((arma::randu<arma::colvec>(4) - 0.5) / 5.0);
+  }
+  /**
+   *
+   * @param state The current State
+   */        
+  bool IsTerminal(const State& state) const
+  {
+    return bool (-cos(state.Theta1())-cos(state.Theta1()
+      + state.Theta2()) > 1.0);
+  }
+  /**
+   * @param state Current State
+   * @param torque Torque Applied 
+   */
+  arma::colvec Dsdt(arma::colvec state,
+               const double torque) const
+  {
+    double m1 = linkMass1;
```

Comment:
Not sure creating an alias for each parameter helps to improve the readability. I guess, we could also just use ``linkLength1 `` instead of ``lc1``. Let me know what you think.

Reply:
I wanted the code to be readable and hence I used the name. I feel the code is more readable to me. However I will alter it as per your suggesstion